packages/confluxscan-api/account.js

```diff
@@ -0,0 +1,218 @@
+      maxTimestamp,
+      withInput,
+      sort,
+    }
+
+    // remove undefined values
+    searchParams = Object.keys(searchParams).reduce((acc, k) => {
+      const v = searchParams[k]
+      if (v !== undefined) acc[k] = v
+      return acc
+    }, {})
+
+    let finished = false
+    let list = []
+    while (!finished) {
+      searchParams.skip = (page - 1) * size
+
+      const res = await fetch(getURL(networkId, 'account', '/transactions'), {
+        searchParams,
+      }).json()
+      if (res.code === 0) list = list.concat(res.data.list)
+      else throw new Error(res.message)
+      if (list.length >= res.data.total) finished = true
+      else page = page + 1
+    }
+    return {list: formatCoreTxs(list, chainId), total: list.length}
+  } catch (err) {
+    return
+  }
+}
+
+function formatEspaceTxs(txs, chainId) {
+  return txs.map(
+    ({
+      blockHash,
+      blockNumber,
+      contractAddress,
+      from,
+      gas,
+      gasPrice,
+      gasUsed,
+      hash,
+      input,
+      isError,
+      nonce,
+      timestamp,
+      to,
+      transactionIndex,
+      txreceipt_status,
+      value,
+    }) => {
+      const tx = {
+        fromScan: true,
+        hash,
+        blockHash,
+        blockNumber,
+        created: parseInt(timestamp, 10) * 1000,
+        extra: {ok: false},
+        payload: {
+          type: '0x0',
+          from,
+          gas,
+          gasPrice: toHex(gasPrice),
+          nonce: toHex(nonce),
+          to,
+          value: toHex(value),
+          chainId,
+        },
+        receipt: {
+          blockHash,
+          blockNumber,
+          gasUsed,
+          transactionIndex: toHex(transactionIndex),
+        },
+      }
+
+      if (isError !== '0' || txreceipt_status !== '1') {
+        tx.status = -1
+        tx.err = 'rejected or canceled'
+      } else tx.status = 5
+      if (contractAddress) tx.receipt.contractCreated = contractAddress
+      if (input && input !== '0x') tx.payload.data = input
+      return tx
+    },
+  )
+}
+
+export async function transactionsESpace({
+  chainId = undefined,
+  networkId = undefined,
+  address = undefined,
+  startblock = undefined,
+  endblock = undefined,
+  page = 1,
+  size = 100,
+  sort = 'desc',
+}) {
+  if (typeof chainId !== 'string') throw new Error('chainId is required')
+  try {
+    let finished = false
+    let list = []
+    while (!finished) {
+      const res = await fetch(getURL(networkId), {
+        searchParams: {
+          module: 'account',
+          action: 'txlist',
+          address,
+          page,
+          offset: size,
+          startblock,
+          endblock,
+          sort: sort.toLowerCase(),
+        },
+      }).json()
+      if (res.status === '1') list = list.concat(res.result)
+      else throw new Error(res.message)
+      if (res.result.length === size) page = page + 1
+      else finished = true
+    }
+    return {
+      list: formatEspaceTxs(list, parseInt(chainId, 16)),
```

Comment:
![CleanShot 2022-07-01 at 09 52 23](https://user-images.githubusercontent.com/16474758/176806888-db8818ef-5877-4976-aeab-6bf841218b58.png)
In `transactionsCoreSpace` method, the `chainId` should be Integer type I think, but in this place, the `chainId` is a hex string?

Reply:
fixed in the latest commit https://github.com/Conflux-Chain/helios/pull/1168/commits/e9bc6f61902e563b1675711fc111bf80d6102779